common/connection.go

```diff
@@ -157,7 +157,12 @@ func NewConnection(
 
 	ctx, cancelCtx := context.WithCancel(ctx)
 
-	conn, _, connErr := wsd.DialContext(ctx, wsURL, header)
+	conn, response, connErr := wsd.DialContext(ctx, wsURL, header)
+	if response != nil {
+		defer func() {
+			_ = response.Body.Close()
```

Comment:
`wsd.DialContext` docs stays: 

```
The response body may not contain the entire response and does not need to be closed by the application.
```

So not sure if we need this :shrug:

Reply:
great but then we need to put a nolint line, this way at least nobody needs to go read what DialContext does and whether for this http.response it needs or does not need to be closed :shrug: